### PR TITLE
Stop service without uv stop

### PR DESCRIPTION
**Pull request by dovholuknf**
get back on the uv loop and clean up. use `uv_cond_wait` to wait until that process completes then exit the main thread after notifying the SCM.

### CODE REVIEW

File: lib/ziti-tunnel-cbs/dns_host.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #define ZITI_TUNNELER_SDK_DNS_HOST_H
 
 #if _WIN32
-#include <windef.h>
+#include <windows.h>
```
Comment (dovholuknf) [inline lib/ziti-tunnel-cbs/dns_host.h:25]: this was needed to fix compilation in Visual Studio

File: programs/ziti-edge-tunnel/instance-config.c [hunk 2]
```diff
@@ -169,9 +169,11 @@ bool save_tunnel_status_to_file() {
 }
 
 void cleanup_instance_config() {
-    ZITI_LOG(TRACE, "Backing up current tunnel status");
+    ZITI_LOG(DEBUG,"Backing up current tunnel status ");
     save_tunnel_status_to_file();
+    ZITI_LOG(DEBUG,"save_tunnel_status_to_file done ");
     if (sem_initialized == 0) {
-        uv_sem_destroy(&sem);
+        //uv_sem_destroy(&sem);
```
Comment (dovholuknf) [inline programs/ziti-edge-tunnel/instance-config.c:176]: this whole block around &sem seemed unnecessary? I don't know why it's there. can i remove it all? @mary-dcouto

File: programs/ziti-edge-tunnel/ziti-edge-tunnel.c [hunk 3]
```diff
@@ -2697,40 +2699,45 @@ void scm_service_run(const char *name) {
...
 }
 
 // called by scm thread, it should not call any uv operations, because all uv operations except uv_async_send are not thread safe
 void scm_service_stop() {
-    stop_tunnel_and_cleanup();
-    close_log();
+    ZITI_LOG(INFO,"stopping via service");
+    uv_mutex_lock(&stop_mutex);
+    ZITI_LOG(DEBUG,"mutex established. sending stop event");
+    ziti_tunnel_async_send(tunneler, scm_service_stop_event, NULL);
```
Comment (dovholuknf) [inline programs/ziti-edge-tunnel/ziti-edge-tunnel.c:2737]: @mary-dcouto this was the key to fixing the svc crash.

File: programs/ziti-edge-tunnel/windows-service.c [hunk 4]
```diff
@@ -239,11 +239,12 @@ VOID SvcInit( DWORD dwArgc, LPTSTR *lpszArgv)
 
     WaitForSingleObject(ghSvcStopEvent, INFINITE);
 
+    ReportSvcStatus(SERVICE_STOP_PENDING, NO_ERROR, 0);
```
Comment (dovholuknf) [inline programs/ziti-edge-tunnel/windows-service.c:242]: added 'pending stop' status too

**Review by mary-dcouto — APPROVED** [review]

**Pull request merged**